src/wheatley.ts

```diff
@@ -656,17 +623,17 @@ export class Wheatley {
         return reply_message;
     }
 
-    is_root(user: Discord.User | Discord.PartialUser | Discord.APIUser): boolean {
-        //return member.roles.cache.some(r => r.id == root_role_id);
-        return root_ids.has(user.id);
+    async check_permissions(
+        options: Discord.GuildMember | Discord.User | Discord.UserResolvable | Discord.FetchMemberOptions,
+        permissions: Discord.PermissionResolvable,
+    ) {
+        const member = await this.try_fetch_guild_member(options);
+        return member?.permissions.has(permissions) ? true : false;
```

Comment:
Maybe
```suggestion
        return !!member?.permissions.has(permissions);
```

Reply:
Does that have some advantage or is that more idiomatic in TypeScript? Otherwise, I don't find double negations particularly readable.

Reply:
`!!` is a somewhat common pattern for turning something like `undefined | boolean` into just `boolean`. Both are fine, the argument for `!!` would be it's more straightforward than the ternary that isn't doing much